Document/0x05d-Testing-Data-Storage.md

```diff
@@ -381,13 +381,13 @@ To determine whether all the `android.util.Log` class' logging functions have be
 Note that the example above only ensures that calls to the Log class' methods will be removed. If the string that will be logged is dynamically constructed, the code that constructs the string may remain in the bytecode. For example, the following code issues an implicit `StringBuilder` to construct the log statement:
 
 ```java
-Log.v("Private key [byte format]: " + key);
+Log.v("Private key tag","Private key [byte format]: " + key);
 ```
 
 The compiled bytecode, however, is equivalent to the bytecode of the following log statement, which constructs the string explicitly:
 
 ```java
-Log.v(new StringBuilder("Private key [byte format]: ").append(key.toString()).toString());
+Log.v(new StringBuilder("Private key tag"),new StringBuilder("Private key [byte format]: ").append(key.toString()).toString());
```

Comment:
```suggestion
Log.v("Private key tag", new StringBuilder("Private key [byte format]: ").append(key.toString()).toString());
```

Reply:
Why using stringbuilder here and not in the line above? Or vice versa

Reply:
Updating could be tricky. I think probably that there could be some leftover in bytecode, which is explained in that paragraph. Check again - tag is not in decompiled code. However log message is, example:
```
 public void sendMessage(View paramView) {
    Intent intent = new Intent((Context)this, DisplayMessageActivity.class);
    EditText editText = (EditText)findViewById(2131165238);
    String str = editText.getText().toString();
    StringBuilder stringBuilder = new StringBuilder("Sending message:");
    stringBuilder.append(editText.toString());
    stringBuilder.toString();
    intent.putExtra("com.example.myfirstapp.MESSAGE", str);
    startActivity(intent);
  }
```
In that case your fix is perfect :) Thanks for spotting.